scripts/triggers-end-to-end-tests/run.spec.js

```diff
@@ -154,6 +160,23 @@ TriggerEndToEndTest.prototype.writeToPubsub = function writeToPubsub(done) {
   return this.invokeHttpFunction("writeToPubsub", done);
 };
 
+TriggerEndToEndTest.prototype.waitForCondition = function(conditionFn, timeout, callback) {
+  let elapsed = 0;
+  let interval = 10;
+  const id = setInterval(() => {
+    if (conditionFn()) {
+      clearInterval(id);
+      callback();
+    }
+
+    elapsed += interval;
+    if (elapsed > timeout) {
+      clearInterval(id);
+      throw new Error(`Timed out waiting for condition: ${JSON.stringify(conditionFn)}`);
```

Comment:
I have a minor preference for calling `callback(new Error(...))` instead of throwing to be consistent with the other `TriggerEndToEndTest` prototype functions, but I'm also fine if change that convention. Just as long as we're consistent.